sync/strategies/simple_onchange.py

```diff
@@ -0,0 +1,42 @@
+from sync.interfaces import IStrategy
+class SimpleOnChangeStrategy(IStrategy):
+    last_seen = {}
+
+    # A strategy that watches emulator memory & sends it when a change is detected
+    # and applies data immediately when it is received (before collecting data to send)
+    def __init__(self, offset_length_pairs):
+        self.offset_length_pairs = offset_length_pairs
+        self.interesting_offsets = list(list(zip(*offset_length_pairs))[0]) # just a list of all the offsets
+
+    def on_emulator_step(self, caller, received_data):
+        self.write_received_data_to_memory(caller, received_data)
+        data_to_send = self.read_memory_for_changes(caller)
+        return data_to_send
+
+    def read_memory_for_changes(self, caller):
```

Comment:
maybe worth remarking in a docstring or comment that this sends *all* the interesting offsets' data, or *none*, never a partial subset.

Reply:
i copied your comment ~